tests/common/ExecutionHelper.cs

```diff
@@ -216,10 +216,15 @@ public bool HasErrorPattern (string prefix, int number, string messagePattern)
 
 		public int WarningCount {
 			get {
-				return messages.Count ((v) => v.IsWarning);
+				return GetWarningCount (messages);
 			}
 		}
 
+		public static int GetWarningCount (List<ToolMessage> messages)
```

Comment:
Minor: if there is no dependency on `List`, consider using `IEnumerable<ToolMessage>`.